lib/compress/zstd_compress_internal.h

```diff
@@ -191,6 +201,12 @@ struct ZSTD_CCtx_s {
 };
 
 
+typedef size_t (*ZSTD_blockCompressor) (
+        ZSTD_matchState_t* ms, seqStore_t* seqStore, U32 rep[ZSTD_REP_NUM],
```

Comment:
Seeing this, I'm wondering if `rep` should be part of `ZSTD_matchState`.
I guess the issue is, `rep` is only updated if the current block is actually compressible, a result which will be discovered only after the block compressor is done. Hence it's included into `blockState` instead.
Maybe there is a need to underline the reasoning behind `blockState` existence directly into the type name.